dubbo-common/src/main/java/org/apache/dubbo/common/url/component/URLParam.java

```diff
@@ -483,7 +483,7 @@ public String getAnyMethodParameter(String key) {
      * @return A new URLParam
      */
     public URLParam addParameter(String key, String value) {
-        if (StringUtils.isEmpty(key) || StringUtils.isEmpty(value)) {
+        if (StringUtils.isEmpty(key)) {
```

Comment:
Why change this? This may cause some usage mistakes

Reply:
When a consumer with a * version invoke a null version provider
Building a remote URL based on the consumer's URL will be contiains * version
It's not a good result and will blocked by PermittedSerializationKeeper.checkSerializationPermitted
This issue may only arise when the * symbol is used and The impact of the change will be widespread
But null valus has meanings seems makes scene and 10+ application running well when I made these changes
It's true that no one dares to say these is no other problem
I've been experimenting for over a month and haven't failed yet

Reply:
And a consumer with a * version invoke a null version provider worked well at 2.7
I use different version and condition-router to achieve gray and business group
And like to make the dubbo:consumer version = *
It's not work when I update to 3.2 or 3.3

Reply:
It would be better to change `PermittedSerializationKeeper.checkSerializationPermitted` in the scenarios.

Reply:
Maybe It's not just this problom
It would be better when the * version consumer called the null version provider to build the null version URL

Reply:
`URL` is a really really basic component for Dubbo, so I would prefer not to change it enless we have the conclusion that we can only change it to fix a issue.